packages/web-extension/src/walletManager/walletManagerWorker.ts

```diff
@@ -58,19 +59,20 @@ export class WalletManagerWorker implements WalletManagerApi {
   }
 
   async activate(props: WalletManagerActivateProps): Promise<void> {
-    const { observableWalletName } = props;
-    if (await this.#isActive(observableWalletName)) {
+    const { walletId } = props;
+    if (this.#isActiveWallet(walletId)) {
       return;
     }
     this.#deactivateWallet();
+    this.#activeWalletId = walletId;
 
-    // Key agent is created by UI script on a per wallet unique channel
+    // Key agent is created by UI script on a per wallet unique channel derived from the keyAgent
     const keyAgent = consumeKeyAgent(
```

Comment:
We could open a single keyAgent channel for all wallets and wait here for the keyAgent with the rightID. This way we could stop having to manage the keyAgent channel open/close cycle.

Reply:
Seems like a good improvement. How complicated would that be to implement?

Reply:
It is an easy change.